src/core/GlobalStyle/mixin.js

```diff
@@ -55,6 +63,28 @@ export const flexColumnCenterAll = css`
   ${flexContainer('center', 'center', 'column')}
 `;
 
+export const TYPOGRAPHY_STYLES = {
+  Headline2_B: [24, 34, 600],
+  Headline3_B: [20, 26, 600],
+  Headline4_B: [16, 26, 600],
+  Headline4_M: [16, 26, 400],
+  Body1_B: [14, 24, 600],
+  Body1_M: [14, 24, 400],
+  Body2_B: [12, 19, 600],
+  Body2_M: [12, 19, 400],
+  Caption_B: [10, 14, 600],
+}; // [font-size, line-height, font-weight]
+
+export const getTypographyStyles = (typography) => {
+  const [fontSize, lineHeight, fontWeight] = TYPOGRAPHY_STYLES[typography];
+  return css`
+    font-size: ${fontSize}px;
+    line-height: ${lineHeight}px;
+    font-weight: ${fontWeight};
+    font-family: ${PRETENDARD_FONT_FAMILY};
+  `;
+};
+
```

Comment:
```suggestion
export const TYPOGRAPHY_STYLES = {
  Headline2_B: {
    fontSize: 20,
    lineHeight: 24,
    fontWeight: 600
  },
  ...
};

export const getTypographyStyles = (typography) => {
  const { fontSize, lineHeight, fontWeight } = TYPOGRAPHY_STYLES[typography];
  return css`
    font-size: ${fontSize}px;
    line-height: ${lineHeight}px;
    font-weight: ${fontWeight};
    font-family: ${PRETENDARD_FONT_FAMILY};
  `;
};

```

이렇게 배열로 구분하는 걸 객체로 바꾸는 게 TYPOGRAPHY_STYLES 자체를 import해서 쓸 때 좋더라구요!! 타입도 좀 수정해 주셔야 할 것 같습니다... ㅠ 감사합니다...

Reply:
index.d.ts에서 TYPOGRAPHY_STYLES에 대한 타입도 export 해주시면 좋을 것 같아요!!

Reply:
@kimjeongwonn
질문) TYPOGRAPHY_STYLES를 바깥에서 import 해서 사용하는 케이스가 어느 경우가 있을까요? `TYPOGRAPHY_STYLES. Headline2_B. fontSize` 이런 식으로 사용되는 케이스가 저는 잘 떠오르지 않아서 사용례가 있는지 여쭤봅니다!

추가로 다른 디자인시스템도 찾아보면서 고민중인데 계속 드는 생각은 제가 찾아본 디자인 시스템들은 대부분 Typography 요소를 Primitive 요소로 제공하고 이런 mixin은 외부로 노출시키지 않는 방식이라서 고민이 되네요. 아예 컴포넌트로 제공하는게 명호님께서 말씀해주신 것처럼 디자인 시스템을 강제시킬 수 있는 가장 좋은 방법인거 같기도 해서요.

> 그리고 제가 만들었던 함수에서는 String을 직접 받아서 해당하는 타이포를 반환하는 방식이었는데 그렇게 하면 컴포넌트의 props으로 string을 넘겨서 바로 적용할 수 있다는 장점을 생각해서 만들었는데... 요렇게 되면 좀 아쉬울 것 같네요 ㅠㅠ

이렇게 말씀하셨던 부분도 컴포넌트를 애초에 만들 때 extends 해서 사용하도록 하면 문제 없을거 같기도 하고요 (https://github.com/dramancompany/remember-ui/pull/49#discussion_r834000695)

Primitive 요소로 쓰지 않을 거라면 저희 컬러 사용하고 있는 것처럼 타이포그래피별 코드 조각을 묶어서 제공하는게 어떤가 싶

### 1. Antd
<img width="748" alt="image" src="https://user-images.githubusercontent.com/35240142/162385040-8ca128da-da1a-4c7a-bd5e-4aa06d9cf9c3.png">

### 2. Mui
<img width="498" alt="image" src="https://user-images.githubusercontent.com/35240142/162385074-f5474602-ef93-44e1-ac32-eb5d62047027.png">

### 3. Chakra ui
<img width="544" alt="image" src="https://user-images.githubusercontent.com/35240142/162385199-76c15938-3ac4-4cf9-b40b-b18d87592d91.png">

### 4. class 101
<img width="719" alt="image" src="https://user-images.githubusercontent.com/35240142/162385301-4ecef99b-5ba6-420c-976d-ca0b37bedad3.png">

### 5. Rebass
<img width="291" alt="image" src="https://user-images.githubusercontent.com/35240142/162385576-46909e6a-5902-456e-aead-0cf423b4df1f.png">

Reply:
오오 개인적으로는 2번 mui 방식으로 variant로 사전 정의된 디자인 적용하고 component로 원하는 태그 넣을 수 있는 수준이 딱 좋지 않을까 싶습니다!